docs/tutorials/byproduct_recovery_determination-solution.ipynb

```diff
@@ -0,0 +1,728 @@
+    "\n",
+    "    - ``prommis.uky.costing.determine_byproduct_recovery``: Implements the byproduct recovery decision framework.\n",
+    " \n",
+    "For guidance on installing these packages, please refer to the Package Installation section at the end of the tutorial."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "ba3193ce-0e23-44d0-a28a-959efe0acb48",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Pyomo packages\n",
+    "from pyomo.environ import (\n",
+    "    ConcreteModel,\n",
+    "    Expression,\n",
+    "    Param,\n",
+    "    TransformationFactory,\n",
+    "    Var,\n",
+    "    value,\n",
+    ")\n",
+    "\n",
+    "# IDAES packages\n",
+    "from idaes.core.util.model_diagnostics import DiagnosticsToolbox\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "\n",
+    "import pytest\n",
+    "\n",
+    "# PrOMMiS packages\n",
+    "from prommis.nanofiltration.diafiltration import (\n",
+    "    main,\n",
+    "    add_costing,\n",
+    "    add_objective,\n",
+    "    add_product_constraints,\n",
+    "    build_model,\n",
+    "    initialize_model,\n",
+    "    print_information,\n",
+    "    set_scaling,\n",
+    "    solve_model,\n",
+    "    unfix_opt_variables,\n",
+    ")\n",
+    "from prommis.uky.costing.costing_dictionaries import load_default_sale_prices\n",
+    "from prommis.uky.costing.determine_byproduct_recovery import (\n",
+    "    ByproductRecovery,\n",
+    "    determine_example_usage,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "25e63a3d-d488-4cd7-9f34-1199a051b851",
+   "metadata": {},
+   "source": [
+    "## __Step 2: Building on the Pre-defined Diafiltration Process__\n",
+    "\n",
+    "This section gives a brief introduction of how to build upon the pre-defined diafiltration process model. The lithium-cobalt diafiltration process is implemented in the file [diafiltration.py](https://github.com/prommis/prommis/blob/main/src/prommis/nanofiltration/diafiltration.py), available in the PrOMMiS repository. \n",
+    "\n",
+    "To construct a process flowsheet using this existing model, users can refer to the detailed walkthrough provided in the [Multi_Stream Contactor Tutorial (Solution)](https://github.com/prommis/prommis/blob/main/docs/tutorials/diafiltration-solution.ipynb). This tutorial outlines major order for building the flowsheet. \n",
+    "\n",
+    "To determine whether lithium and cobalt should be recovered, the user first needs to __build the diafiltration process__ and __access the cost and revenue__.\n",
+    "\n",
+    "Refer the process model defined in ``diafiltration.py``. Specifically, call the functions in the order specified within the ``main()`` function. Following this order is crucial because the diafiltration process consists of three membrane stages, and the model is built stage by stage. \n",
```

Comment:
```suggestion
    "Refer to the process model defined in ``diafiltration.py``. Specifically, call the functions in the order specified within the ``main()`` function. Following this order is crucial because the diafiltration process consists of three membrane stages, and the model is built stage by stage. \n",
```